src/sharepoint/rest/rest.ts

```diff
@@ -1,10 +1,75 @@
 "use strict";
 
-import { Web } from "./web";
+import { Site } from "./site";
+import { Web } from "./webs";
+import * as Util from "../../utils/util";
+import { Queryable } from "./queryable";
 
 /**
  * Root of the SharePoint REST module
  */
 export class Rest {
-    public web = new Web("_api");
+
+    /**
+     * Begins a site collection scoped REST request
+     * 
+     * @param url The base url for the request, optional if running in the context of a page
+     */
+    public get site(): Site {
+        return new Site("_api");
+    }
+
+    /**
+     * Begins a web scoped REST request
+     * 
+     * @param url The base url for the request, optional if running in the context of a page
+     */
+    public get web(): Web {
+        return new Web("_api");
+    }
+
+    /**
+     * Begins a cross-domain, host site scoped REST request, for use in add-in webs
+     * 
+     * @param addInWebUrl The absolute url of the add-in web
+     * @param hostWebUrl The absolute url of the host web
+     */
+    public crossDomainSite(addInWebUrl: string, hostWebUrl: string): Site {
+        return this._cdImpl<Site>(Site, addInWebUrl, hostWebUrl, "site");
+    }
+
+    /**
+     * Begins a cross-domain, host web scoped REST request, for use in add-in webs
+     * 
+     * @param addInWebUrl The absolute url of the add-in web
+     * @param hostWebUrl The absolute url of the host web
+     */
+    public crossDomainWeb(addInWebUrl: string, hostWebUrl: string): Web {
```

Comment:
Hmm.. Wouldn't it be more logical to specify which context you are working against in the configuration and have that new up the correct implementation so that you can move things out from an add-in with easy?

I.e there should only be one web() method